gateway/host_checker.go

```diff
@@ -51,6 +49,11 @@ type HostHealthReport struct {
 	IsTCPError   bool
 }
 
+type HostSample struct {
```

Comment:
Why you decided to add a struct, instead of just using counter as previously?
Why we need a new `reachedLimit` field?

Reply:
@buger I did it this way so we can delete `unHealthyList` and avoid having 2 sources of the truth of which host is down / up.

If we don't have some kind of flag / list that determines that a host reached the trigger limit (which was basically what `unHealthyList` did), the "host up" will start decreasing the counter number even if it doesn't reach the trigger limit.
This can cause that a host which is responding 1 down, 1 up with a trigger limit 2 never reach the trigger limit and will never be removed from the round robin.